src/PatternMatcher.php

```diff
@@ -62,6 +62,8 @@ private function isMatch(Pattern $pattern, string $filename): bool
                 '*' => '[^\/]+',
                 '?' => '[^\/]',
                 '**' => '.*',
+                '/**' => '.*',
```

Comment:
```suggestion
                '/**' => '\/.*',
```

I think we do need to add the slashes. I have created a pull request on a test repository to verify some edge cases. I have added a coworker as codeowner to that repository: https://github.com/timoschinkel/test-repository/pull/6/files

For the rule `docs/**` that does apply to `docs/file.md` and `docs/folder/file.md`, but not to `docs.md`. 

These situations can be tested by adding the following lines to `\CodeOwners\Tests\PatternMatcherTest::provideNoMatchFoundExceptionIsThrownForFilename()`:

```php
            // **
            [new Pattern('**/foo', ['@owner']), 'foo.ext'],
            [new Pattern('foo/**', ['@owner']), 'foo.ext'],
```